### ISSUE TITLE
Pronunciation vs. Description

### ISSUE THREAD

**Issue by nathanhammond** [lshk-org/jyutping-table#3]
There are numerous fields in this TSV which feature extractions from the 10646hk JPFullTable.pdf document in the 《部件名稱表》 section. It is not clear to me that *descriptive* values (generally of a radical, but also sometimes to distinguish between identically sounding but functionally distinct characters) should be included in this list–especially since it is designed for *computers*.

For example:
```tsv
氵	U+6C35	saam1 dim2 seoi2
```

Presented with that character *component* in a text, a situation which occurs almost exclusively in an educational setting, maybe in a sentence such as "譬如話氵用個字倒鎖會同水有關", a reader may indeed describe that character in that way, but that's an expansion to `三點水`.

In English, I don't ascribe a pronunciation to the the "right-hand ascender of the lowercase letter d" as first, it doesn't have one, and second, the way I identified it is descriptive.

Given that, I would say that `氵`, presented as a standalone character, either has no pronunciation or has a pronunciation of `seoi2`.

***

Here is a list of cases in `list.tsv` which appear to have descriptive values.

```tsv
㓁	U+34C1	tuk1 tau4 jyut6
㔾	U+353E	ngau1 zi6 pong4
㖊	U+358A	jing1 cam4
𥪕	U+25A95	cin1 sing1
𦉫	U+2626B	hoi1 hau2 ming5
𠃑	U+200D1	saam1 zit3 syu6
𠃍	U+200CD	waang4 syu6 gok3
𠺖	U+20E96	jing1 mau5
𠰴	U+20C34	jing1 sek6
𤣩	U+248E9	juk6 wong4 bin1
𦉪	U+2626A	hoi1 hau2 sei3
𠂊	U+2008A	kyut3 naat6 gau2
𧘇	U+27607	ji1 zi6 dai2
𡯂	U+21BC2	baat3 gei1
𠯢	U+20BE2	saa1 aa6
飠	U+98E0	sik6 zi6 bin1
饣	U+9963	gaan2 faa3 sik6 zi6 bin1
髟	U+9ADF	faat3 zi6 tau4
阝	U+961D	ji5 zai2 bin1
钅	U+9485	gaan2 faa3 gam1 zi6 bin1
釒	U+91D2	gam1 zi6 bin1
辶	U+8FB6	caang1 teng5
覀	U+8980	sai1 zi6 tau4
讠	U+8BA0	gaan2 faa3 jin4 zi6 bin1
衤	U+8864	ji1 zi6 bin1
虍	U+864D	fu2 zi6 tau4
艹	U+8279	cou2 faa1 tau2
罒	U+7F52	sei3 mong5 tau4
罓	U+7F53	gaau1 caa1 sei3
耂	U+8002	lou5 zi6 tau4
肀	U+8080	syu1 zi6 tau4
籿	U+7C7F	fan1 mai5
粀	U+7C80	sap6 mai5
粁	U+7C81	cin1 mai5
粌	U+7C8C	baak3 mai5
粍	U+7C8D	hou4 mai5
粨	U+7CA8	baak3 mai5
粴	U+7CB4	gung1 lei5
糎	U+7CCE	lei4 mai5
糹	U+7CF9	kiu5 si1 bin1
纟	U+7E9F	gaan2 faa3 kiu5 si1 bin1
礻	U+793B	si6 zi6 bin1
禸	U+79B8	maan6 zi6 dai2
竍	U+7ACD	sap6 sing1
竏	U+7ACF	cin1 sing1
竓	U+7AD3	hou4 sing1
竕	U+7AD5	fan1 sing1
竡	U+7AE1	baak3 sing1
竰	U+7AF0	lei4 sing1
瓧	U+74E7	sap6 ngaa5
瓩	U+74E9	cin1 ngaa5
瓰	U+74F0	fan1 ngaa5
瓱	U+74F1	hou4 ngaa5
瓸	U+74F8	baak3 ngaa5
甅	U+7505	lei4 ngaa5
疒	U+7592	beng6 zi6 tau4
癶	U+7676	dang1 zi6 tau4
爫	U+722B	zaau2 zi6 tau4
牜	U+725C	zo2 pong4 ngau4
犭	U+72AD	gau2 zaau2 bin1
灬	U+706C	sei3 dim2
㣺	U+38FA	zang1 dim2 syu6 sam1 bin1
䇆	U+41C6	sap6 sing1
䇉	U+41C9	baak3 sing1
䒑	U+4491	bing6 zi6 tau4
丆	U+4E06	jat1 waang4 jat1 pit3
丬	U+4E2C	gaan2 baan6 bin1
丷	U+4E37	baat3 zi6 tau4
乄	U+4E44	ngau1 mei5 gaau1 caa1
乊	U+4E4A	fu4 zi6 tau4
乛	U+4E5B	waang4 ngau1
亠	U+4EA0	jat1 dim2 jat1 waang4
亻	U+4EBB	kei5 jan4 bin1
儿	U+513F	gaan2 faa3 ji4
兙	U+5159	sap6 hak1
兛	U+515B	cin1 hak1
兝	U+515D	fan1 hak1
兞	U+515E	hou4 hak1
兡	U+5161	baak3 hak1
兣	U+5163	lei4 hak1
冂	U+5182	tung4 zi6 kwaang1
冫	U+51AB	loeng5 dim2 seoi2
凵	U+51F5	hung1 zi6 kwaang1
刂	U+5202	lei6 dou1 bin1
勹	U+52F9	baau1 zi6 tau4
匚	U+531A	hong1 zi6 kwaang1
卅	U+5345	saa1 aa6
卌	U+534C	sei3 aa6
卩	U+5369	daan1 ji5 dou1
吋	U+540B	jing1 cyun3
呎	U+544E	jing1 cek3
哩	U+54E9	jing1 lei5
唡	U+5521	jing1 loeng2
啢	U+5562	jing1 loeng2
嗧	U+55E7	gaa1 leon4
噚	U+565A	jing1 cam4
夂	U+5902	pit3 pong4 jau6
夊	U+590A	pit3 soek3 jau6
宀	U+5B80	lam1 pung4 tau2
尢	U+5C22	jau4 zi6 pong4
尣	U+5C23	baat3 jan4
屮	U+5C6E	bun3 bin1 cou2
广	U+5E7F	gwong2 zi6 tau4
廴	U+5EF4	gin3 zi6 pong4
廾	U+5EFE	lung6 zi6 dai2
彐	U+5F50	gaan2 faa3 luk6 zi6 tau4
彑	U+5F51	kyut3 tau4 wu6
彡	U+5F61	saam1 pit3
彳	U+5F73	soeng1 kei5 jan2
忄	U+5FC4	syu6 sam1 bin1
扌	U+624C	tik1 sau2 bin1
攵	U+6535	faan2 man4 bin1
氵	U+6C35	saam1 dim2 seoi2
氺	U+6C3A	taai3 zi6 dai2
浬	U+6D6C	hoi2 lei5
```

**Comment by nathanhammond** [comment]
Here is a complete list of the fields which contain descriptive pronunciations from the 粵語拼音表 portion of the list.

After further research, these appear to likely be archaic contractions that didn't make it to modern Cantonese. Some of them may be bad handwriting or [bad transcriptions](https://www.dampfkraft.com/ghost-characters.html) (e.g. `䇉 U+41C9 baak3 sing1` could have been intended to be `竡`).

The ones from the 部件名稱表 table, however, are descriptive and definitively *not* contractions. I still believe those should be removed from this list.

```tsv
籿	U+7C7F	fan1 mai5
粀	U+7C80	sap6 mai5
粁	U+7C81	cin1 mai5
粌	U+7C8C	baak3 mai5
粍	U+7C8D	hou4 mai5
粨	U+7CA8	baak3 mai5
糎	U+7CCE	lei4 mai5

㖊	U+358A	jing1 cam4
吋	U+540B	jing1 cyun3
呎	U+544E	jing1 cek3
哩	U+54E9	jing1 lei5
唡	U+5521	jing1 loeng2
啢	U+5562	jing1 loeng2
噚	U+565A	jing1 cam4
𠺖	U+F45A	jing1 mau5
𠰴	U+F4C0	jing1 sek6

䇆	U+41C6	sap6 sing1
䇉	U+41C9	baak3 sing1
竍	U+7ACD	sap6 sing1
竏	U+7ACF	cin1 sing1
竓	U+7AD3	hou4 sing1
竕	U+7AD5	fan1 sing1
竡	U+7AE1	baak3 sing1
竰	U+7AF0	lei4 sing1
𥪕	U+ED41	cin1 sing1

兙	U+5159	sap6 hak1
兛	U+515B	cin1 hak1
兝	U+515D	fan1 hak1
兞	U+515E	hou4 hak1
兡	U+5161	baak3 hak1
兣	U+5163	lei4 hak1

𠯢	U+E064	saa1 aa6
〺	U+5345	saa1 aa6
卌	U+534C	sei3 aa6

瓧	U+74E7	sap6 ngaa5
瓩	U+74E9	cin1 ngaa5
瓰	U+74F0	fan1 ngaa5
瓱	U+74F1	hou4 ngaa5
瓸	U+74F8	baak3 ngaa5
甅	U+7505	lei4 ngaa5

浬	U+6D6C	hoi2 lei5
粴	U+7CB4	gung1 lei5

嗧	U+55E7	gaa1 leon4
```

**Comment by chaaklau** [comment]
> Presented with that character _component_ in a text, a situation which occurs almost exclusively in an educational setting, maybe in a sentence such as "譬如話氵用個字倒鎖會同水有關", a reader may indeed describe that character in that way, but that's an expansion to `三點水`.

@nathanhammond  As you have correctly pointed out, many of these characters are *components*, which are almost never used in normal writing. I can think of two contexts in which these characters will be found:  
1. Education setting (e.g. explanation of character structures in language textbooks)
2. An intended character is not supported by Unicode / the keyboard / the font, therefore users split the characters into two (e.g. 亻因 Taiwanese Hokkien word for "they"). There can also be other reasons for using components instead of the original characters (for fun, to bypass censorship, etc.).

For `1`, in education setting, the polysyllabic pronunciation is expected. E.g. I would expect a TTS engine to read  `氵` as  `三點水` / `saam1 dim2 seoi2`. This particular one is quite stable, there is no other name for this component. For some lesser-used components, there may not be a unified name.

----

> After further research, these appear to likely be archaic contractions that didn't make it to modern Cantonese. Some of them may be bad handwriting or bad transcriptions (e.g. 䇉 U+41C9 baak3 sing1 could have been intended to be 竡).

I can confirm that unit-related "contractions/combined characters" are not used in Modern Cantonese. Out of all polysyllabic entries, I believe only this is in common use.
```
〺	U+5345	saa1 aa6
```
----
I agree that the _descriptive entries_ should be removed from the table. Many of these components do not have a fixed name, and the values from the original file are just one way of expanding them. We can split `list.tsv` into two files, one for true character-to-jyutping mapping, and one for descriptive values. 

These are true contractions (numbers) and should be kept.
```
𠯢	U+E064	saa1 aa6
〺	U+5345	saa1 aa6
卌	U+534C	sei3 aa6
```

These archaic combined characters should be kept. 
```
籿	U+7C7F	fan1 mai5
粀	U+7C80	sap6 mai5
粁	U+7C81	cin1 mai5
粌	U+7C8C	baak3 mai5
粍	U+7C8D	hou4 mai5
粨	U+7CA8	baak3 mai5
糎	U+7CCE	lei4 mai5

䇆	U+41C6	sap6 sing1
䇉	U+41C9	baak3 sing1
竍	U+7ACD	sap6 sing1
竏	U+7ACF	cin1 sing1
竓	U+7AD3	hou4 sing1
竕	U+7AD5	fan1 sing1
竡	U+7AE1	baak3 sing1
竰	U+7AF0	lei4 sing1
𥪕	U+ED41	cin1 sing1

兙	U+5159	sap6 hak1
兛	U+515B	cin1 hak1
兝	U+515D	fan1 hak1
兞	U+515E	hou4 hak1
兡	U+5161	baak3 hak1
兣	U+5163	lei4 hak1

瓧	U+74E7	sap6 ngaa5
瓩	U+74E9	cin1 ngaa5
瓰	U+74F0	fan1 ngaa5
瓱	U+74F1	hou4 ngaa5
瓸	U+74F8	baak3 ngaa5
甅	U+7505	lei4 ngaa5

嗧	U+55E7	gaa1 leon4
```

I feel that these are also descriptions and should be removed from the main table.
```
浬	U+6D6C	hoi2 lei5
粴	U+7CB4	gung1 lei5

㖊	U+358A	jing1 cam4
吋	U+540B	jing1 cyun3
呎	U+544E	jing1 cek3
哩	U+54E9	jing1 lei5
唡	U+5521	jing1 loeng2
啢	U+5562	jing1 loeng2
噚	U+565A	jing1 cam4
𠺖	U+F45A	jing1 mau5
𠰴	U+F4C0	jing1 sek6
```
----
I also read the linked issue. After the change, the Jyutping field will continue to contain polysyllabic entries. Also note that more combined characters are being created by Cantonese users, some of them may make their ways to Unicode.

**Comment by nathanhammond** [comment]
Thank you for your detailed response! I appreciate having an additional person review this. Some updates:

I'm definitely not afraid of having to write code to handle polysyllabic characters; I've already got to support `〺`. If I have to handle polysyllabic characters even once I may as well do it right. (I'm more in the descriptivist camp when it comes to language.) Thinking about it more, for the archaic "contraction" forms, since all but 6 do not collide with any other meanings or pronunciations there is no particular value in omitting them. (That also gets closer to prescriptivism, which on principle I want to oppose.)

For the six characters that have both monosyllabic and polysyllabic pronunciations listed in 10646HK I can always choose monosyllabic and almost always be correct.

```tsv
ch	ucs2	pronunciation
吋	U+540B	cyun3
呎	U+544E	cek3
哩	U+54E9	(li1, le1, lei5)
浬	U+6D6C	lei5
𠺖	U+F45A	mau5
𠰴	U+F4C0	sek6
```

***

For descriptive readings, I find that should likely be a separate field. A character can have a "pronunciation" or a "description" —or even both. I consider this to be similar to how each Unicode character receives a name. For example, `氵`, `三點水` is also known as `CJK Radical Water One`. I absolutely agree that `三點水` is how that character component should be described in Cantonese, but only if "descriptive readings" (probably a default "on" option) is set in the TTS.

***

In [my implementation of extracting data from 10646HK](https://github.com/nathanhammond/parse-jyutping-table-full) I've updated it to process each of the tables separately. We now have two separate implementations which produce the same output which gives me confidence that we have successfully captured the source data without error.

I'm in the process of comparing that data to the contents of this repository. I'll have more comments on that soon. (I need to have a better understanding of all the possible character sets involved beforehand.)

***

Given all of this, I would like to propose that this be addressed by:
- Creating an additional field for descriptions (not a separate list; a description field has value and belongs in the list).
- Placing descriptive values in the descriptive field and dropping the value from the pronunciation field.
